spidermon/contrib/scrapy/monitors.py

```diff
@@ -174,6 +174,99 @@ def test_verify_item_validation_error(self):
         self.assertTrue(item_validation_errors <= errors_threshold, msg=msg)
 
 
+@monitors.name("Field Coverage Monitor")
+class FieldCoverageMonitor(BaseScrapyMonitor):
+    """Validate if field coverage rules are met.
+
+    To use this monitor you need to enable ``SPIDERMON_ADD_FIELD_COVERAGE``
+    setting that will add information about field coverage on your spider
+    statistics.
+
+    To define your field coverage rules create a dictionary containing the
+    expected coverage for each field you want to monitor.
+
+    As example, if the items you are returning from your spider are a Python dictionaries
+    with the following format:
+
+    .. code-block:: python
+
+        {
+            "field_1": "some_value",
+            "field_2": "some_value",
+            "field_3": {
+                "field_3_1": "some_value",
+                "field_3_2": "some_value",
+            }
+        }
+
+    A set of rules may be defined as follows:
+
+    .. code-block:: python
+
+        # project/settings.py
+        SPIDERMON_FIELD_COVERAGE_RULES = {
+            "dict/field_1": 0.4,  # Expected 40% coverage for field_1
+            "dict/field_2": 1.0,  # Expected 100% coverage for field_2
+            "dict/field_3": 1.0,  # Expected 100% coverage for parent field_3
+            "dict/field_3/field_3_1": 0.5,  # Expected 50% coverage for nested field_3_1
+        }
+
+    You are not obligated to set rules for every field, just the ones thar you are interested.
+    Also, you can monitor nested fields if available in your returned items.
+
+    .. note::
+       If you are returning an item type other than a dictionary, replace `dict` by the
+       class name of the item you are returning.
+
+       Considering you have an item defined as:
+
+       .. code-block:: python
+
+           class MyCustomItem(scrapy.Item):
+               field_1 = scrapy.Field()
+               field_2 = scrapy.Field()
+
+       You must define the field coverage rules as follows:
+
+       .. code-block:: python
+
+           SPIDERMON_FIELD_COVERAGE_RULES = {
+               "MyCustomItem/field_1": 0.4,
+               "MyCustomItem/field_2": 1.0,
+           }"""
+
+    def run(self, result):
+        add_field_coverage_set = self.crawler.settings.getbool(
+            "SPIDERMON_ADD_FIELD_COVERAGE", False
+        )
+        if not add_field_coverage_set:
+            raise NotConfigured(
+                "You should enable field coverage stats (SPIDERMON_ADD_FIELD_COVERAGE=True) on your project settings"
```

Comment:
I’m not completely sure about the context where this appears to the user, as I’m not familiar with how Spidermon monitors are enabled and configured, so my suggestion below may be completely wrong.

But, assuming that this message appears when users have actively added the monitor somewhere, but might either have forgotten to set this setting or may have set this setting to `False` on purpose (e.g. to temporarily disable the monitor without removing it from where they added it), I would avoid this “should” phrasing.

```suggestion
                "To enable field coverage stats, set SPIDERMON_ADD_FIELD_COVERAGE=True in your project settings"
```

Reply:
This setting is needed to enable the monitor, not the stats. I changed to `"To enable field coverage monitor, set SPIDERMON_ADD_FIELD_COVERAGE=True in your project settings"` so it is clearer what is the purpose of this setting.